tests/ext/drv/test_ext_drv.c

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+/*++
+
+Abstract:
+WDF based driver that does the following:
+
+Registers as an eBPF program information provider and hook provider.
+
+Environment:
+
+    Kernel mode
+
+--*/
+
+// ntddk.h needs to be included first due to inter header dependencies on Windows.
+#include <ntddk.h>
+
+#pragma warning(push)
+#include <wdf.h>
+
+#include "ebpf_platform.h"
+
+#include "test_ext.h"
+#include "test_ext_helpers.h"
+#include "test_ext_ioctls.h"
+
+#define TEST_EBPF_EXT_DEVICE_NAME L"\\Device\\TestEbpfExt"
+#define TEST_EBPF_EXT_SYMBOLIC_DEVICE_NAME L"\\GLOBAL??\\TestEbpfExtIoDevice"
+
+// Driver global variables
+static DEVICE_OBJECT* _test_ebpf_ext_driver_device_object;
+static BOOLEAN _test_ebpf_ext_driver_unloading_flag = FALSE;
+
+//
+// Pre-Declarations
+//
+static EVT_WDF_IO_QUEUE_IO_DEVICE_CONTROL _test_ebpf_ext_driver_io_device_control;
+DRIVER_INITIALIZE DriverEntry;
+
+static VOID
+_test_ebpf_ext_driver_io_device_control(
+    _In_ WDFQUEUE queue,
+    _In_ WDFREQUEST request,
+    size_t output_buffer_length,
+    size_t input_buffer_length,
+    ULONG io_control_code);
```

Comment:
can you use a standard type here like uint32_t or will the compiler complain?

Reply:
compiler does not like it, likely due to https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdfio/nc-wdfio-evt_wdf_io_queue_io_device_control